app/controllers/productController.js

```diff
@@ -40,6 +44,42 @@ const productController = {
         message: "Internal server error"
       })
     }
+  },
+  getAllProduct: async (req, res) => {
+    try {
+      const { product, category, page } = req.query
+
+      const pageNumber = page ? parseInt(page, 10) : 1
+
+      if (isNaN(pageNumber) || pageNumber < 1) {
```

Comment:
kurang convert to number

Reply:
itu hasilnya udh di tabrak di bawah mi

Reply:
ketika dia masuk string udh di handle langsung

Reply:
oh oke, udah pake parseInt ya